server/src/session.ts

```diff
@@ -1164,41 +1167,47 @@ export class Session {
     this.disableLanguageServiceForProject(project, 'ngcc is running');
     const configFilePath = project.getConfigFilePath();
 
-    this.connection.sendProgress(NgccProgressType, NgccProgressToken, {
-      done: false,
-      configFilePath,
-      message: `Running ngcc for ${configFilePath}`,
-    });
+    const progressReporter = await this.connection.window.createWorkDoneProgress();
 
-    let success = false;
+    progressReporter.begin('Angular', 0, `Running ngcc for ${configFilePath}`);
 
     try {
       this.projectNgccQueue.push({project, done: false});
       await resolveAndRunNgcc(configFilePath, {
         report: (msg: string) => {
-          this.connection.sendProgress(NgccProgressType, NgccProgressToken, {
-            done: false,
-            configFilePath,
-            message: msg,
-          });
+          progressReporter.report(50, msg);
         },
       });
-      success = true;
     } catch (e) {
       this.error(
           `Failed to run ngcc for ${
               configFilePath}, language service may not operate correctly:\n` +
           `    ${e.message}`);
+
+      this.connection.window
+          .showErrorMessage<{
+            action: NgccErrorMessageAction,
+            title: string,
+          }>(`Angular extension might not work correctly because ngcc operation failed. ` +
+                 `Try to invoke ngcc manually by running 'npm/yarn run ngcc'. ` +
+                 `Please see the extension output for more information.`,
+             {title: 'Show output', action: NgccErrorMessageAction.showOutput})
+          .then((selection) => {
+            switch (selection?.action) {
+              case NgccErrorMessageAction.showOutput:
+                this.connection.sendNotification(OpenOutputChannel, {});
+                break;
+            }
```

Comment:
nit: given that there is only 1 option here, I think this would be best written as an `if`
```suggestion
            if (selection?.action === NgccErrorMessageAction.showOutput) {
                this.connection.sendNotification(OpenOutputChannel, {});
            }
```

This can always change later if we really do have a bunch of options, but I don't think that's likely to happen any time soon.